### PR TITLE
Add learn api

### PR DESCRIPTION
**Pull request by aradoje**

### CODE REVIEW

File: src/Api/Learn.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+     *
+     * @return ResponseInterface|Accepted
+     */
+    public function create(string $candidate, string $role, string $learnName, string $category, ?int $score = null)
+    {
+        Assert::notEmpty($candidate, 'Candidate cannot be empty');
+        Assert::notEmpty($role, 'Role cannot be empty');
+        Assert::notEmpty($learnName, 'Learn name cannot be empty');
+        Assert::notEmpty($category, 'Category cannot be empty');
+        Assert::oneOf($category, [
```
Comment (Nyholm) [inline src/Api/Learn.php:28]: We should let the API handle this instead.

I suggest moving these categories to the doc block instead.

File: src/Api/Learn.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            'soft_performance_data',
+            ],
+        'Invalid category');
+
+        $response = $this->httpPost('/api/learn', [
+            'candidate' => $candidate,
+            'role' => $role,
+            'name' => $learnName,
+            'category' => $category,
+            'score' => $score ?? 0,
```
Comment (Nyholm) [inline src/Api/Learn.php:44]: Using `0` as "score is missing" is wrong. As missing is not the same as `0`.

File: src/Api/Learn.php (hunk 1 above)
Comment (aradoje) [reply src/Api/Learn.php:28]: I'll remove them then, since we have the allowed values in the docs

Comment (Nyholm) [reply src/Api/Learn.php:28]: Yeah, but if you put them in the doc block you can get IDE autocompletion.

**Pull request merged**